core/types/state_account_storage.libevm_test.go

```diff
@@ -51,72 +51,68 @@ func TestStateAccountExtraViaTrieStorage(t *testing.T) {
 		arbitrary   = common.HexToHash("0x94eecff1444ab69437636630918c15596e001b30b973f03e06006ae20aa6e307")
 	)
 
+	// An assertion is the actual test to be performed. It is returned upon
+	// registration, instead of being a standalone field in the test, because
+	// each one uses a different generic parameter.
+	type assertion func(*testing.T, *types.StateAccount)
 	tests := []struct {
 		name                string
-		registerAndSetExtra func(*types.StateAccount) *types.StateAccount
-		assertExtra         func(*testing.T, *types.StateAccount)
+		registerAndSetExtra func(*types.StateAccount) (*types.StateAccount, assertion)
 		wantTrieHash        common.Hash
 	}{
 		{
 			name: "vanilla geth",
-			registerAndSetExtra: func(a *types.StateAccount) *types.StateAccount {
-				return a
-			},
-			assertExtra: func(t *testing.T, a *types.StateAccount) {
-				t.Helper()
-				assert.Truef(t, a.Extra.Equal(nil), "%T.%T.IsEmpty()", a, a.Extra)
+			registerAndSetExtra: func(a *types.StateAccount) (*types.StateAccount, assertion) {
+				//nolint:thelper // It's more useful if this test failure shows this line instead of its call site
+				return a, func(t *testing.T, got *types.StateAccount) {
+					assert.Truef(t, a.Extra.Equal(nil), "%T.%T.IsEmpty()", a, a.Extra)
+				}
 			},
 			wantTrieHash: vanillaGeth,
 		},
 		{
 			name: "true-boolean payload",
-			registerAndSetExtra: func(a *types.StateAccount) *types.StateAccount {
-				types.RegisterExtras[bool]().SetOnPayloadCarrier(a, true)
-				return a
-			},
-			assertExtra: func(t *testing.T, sa *types.StateAccount) {
-				t.Helper()
-				assert.Truef(t, types.ExtraPayloads[bool]{}.FromPayloadCarrier(sa), "")
+			registerAndSetExtra: func(a *types.StateAccount) (*types.StateAccount, assertion) {
+				e := types.RegisterExtras[bool]()
+				e.SetOnPayloadCarrier(a, true)
```

Comment:
should the calls in this file to `SetOnPayloadCarrier` be modified to the new StateAccount.Set?

Reply:
I thought the linter would catch these. Removing the functions will force it anyway.